src/lib/containers/entity_finder/tree/EntityTree.tsx

```diff
@@ -171,96 +164,112 @@ export function EntityTree(props: EntityTreeProps) {
       : { sort: 'PROJECT_NAME', sortDirection: 'ASC' },
     {
       enabled: useProjectData,
-      // Don't refetch the projects. Updating the entity headers will drop all of the children that VirtualizedTree has fetched
+      // Don't refetch the projects. Updating the entity headers will drop all the children that VirtualizedTree has fetched
       refetchInterval: Infinity,
     },
   )
 
   const { data: currentContainerBundle, isSuccess: isSuccessBundle } =
     useGetEntityBundle(currentContainer!, undefined, ALL_ENTITY_BUNDLE_FIELDS, {
       enabled: !!currentContainer && currentContainer !== 'root',
+      useErrorBoundary: true,
     })
 
-  useEffect(() => {
-    if (useProjectData && isSuccessProjects) {
-      if (projectData?.pages) {
-        setTopLevelEntities(
-          projectData.pages.flatMap(page => page.results).map(toEntityHeader),
-        )
-      }
-    }
-  }, [useProjectData, isSuccessProjects, projectData])
+  const { data: favorites, isLoading: isLoadingFavorites } = useGetFavorites(
+    'NAME',
+    'ASC',
+    {
+      select: data => ({
+        // TODO: https://sagebionetworks.jira.com/browse/PLFM-6652
+        ...data,
+        results: data.results.filter(eh =>
+          visibleTypes.includes(convertToEntityType(eh.type)),
+        ),
+      }),
+      // Don't refetch the projects. Updating the entity headers will drop all the children that VirtualizedTree has fetched
+      refetchInterval: Infinity,
+      useErrorBoundary: true,
+    },
+  )
+
+  const { data: initialContainerPath } = useGetEntityPath(initialContainer!, {
+    enabled: !!(
+      projectId &&
+      initialContainer &&
+      initialContainer.match(SYNAPSE_ENTITY_ID_REGEX)
+    ),
+    refetchInterval: Infinity,
+    useErrorBoundary: true,
+  })
+
+  const {
+    data: projectHeader,
+    isLoading: isLoadingProjectHeader,
+    error: fetchProjectError,
+  } = useGetEntityHeader(projectId!, {
+    enabled: !!(projectId ?? initialContainerPath?.path[1]?.id),
+    refetchInterval: Infinity,
+  })
+
+  if (
+    FinderScope.CURRENT_PROJECT === scope &&
+    projectId &&
+    fetchProjectError &&
+    fetchProjectError.status === 403
+  ) {
+    // The user doesn't have access to the project, so let's change the scope to something else
+    displayToast(
+      `You don't have access to the current project (${projectId}).`,
+      'warning',
+    )
+    setScope(FinderScope.CREATED_BY_ME)
+  }
 
   // Populates the first level of entities in the tree view
-  useEffect(() => {
-    setIsLoading(true)
+  const {
+    topLevelEntities,
+    isLoading,
+  }: {
+    topLevelEntities: Pick<EntityFinderHeader, 'name' | 'id' | 'type'>[]
+    isLoading: boolean
+  } = useMemo(() => {
```

Comment:
Remove `useEffect` calling `setState` and replace with `useMemo` deriving state from existing values